frontend/src/components/app/app.tsx

```diff
@@ -1,26 +1,43 @@
 import { FC } from 'common/types/types';
-import { MainPage } from 'pages/main-page';
 import { AppRoute } from 'common/enums/enums';
-import { Routes, Route, HeaderContainer } from 'components/common/common';
-import { useLocation } from 'react-router-dom';
+import { useLocation, useAppSelector, useEffect, useAppDispatch } from 'hooks/hooks';
+import { tokensStorageService } from 'services/services';
+import { authActions } from 'store/actions';
+import { MainPage } from 'pages/main-page';
+import { Routes, Route, HeaderContainer, SidebarContainer } from 'components/common/common';
+import { RestorePasswordPage, SignInPage, SignUpPage } from 'components/auth/auth';
+import { Studio, StudioAnalytics } from 'components/studio';
 import { Search } from 'components/search/search';
-import { SidebarContainer } from 'components/common/sidebar/sidebar-container';
+import { NotFound } from 'components/not-found-page/not-found';
+import { ReactNotifications } from 'react-notifications-component';
 import { isRouteHaveHeader } from 'helpers/routes/is-route-have-header';
-import { NotFound } from '../not-found-page/not-found';
 import { ConfirmationModalTest } from './tests/confirmation-modal/confirmation-modal';
-import { Studio, StudioAnalytics } from '../studio';
-import { RestorePasswordPage, SignInPage, SignUpPage } from 'components/auth/auth';
 import { VideoCardTest } from './tests/video-card/video-card';
 import { VideoPageContainer } from 'pages/video/video-page-container';
 
 import styles from './app.module.scss';
 
 const App: FC = () => {
+  const dispatch = useAppDispatch();
   const { pathname } = useLocation();
+  const { user } = useAppSelector((state) => ({
+    user: state.auth.user,
+  }));
+
+  const hasToken = Boolean(tokensStorageService.getTokens().accessToken);
+  const hasUser = Boolean(user);
+
   const isHaveHeader = isRouteHaveHeader(pathname);
 
+  useEffect(() => {
+    if (hasToken) {
+      dispatch(authActions.loadCurrentUser());
+    }
+  }, [hasToken, hasUser, dispatch]);
+
```

Comment:
if I understand correctly, this dispatch will work twice (user is null initially, then the user changes in the reducer and useEffect works twice)